include/cjdb/contracts.hpp

```diff
@@ -4,10 +4,46 @@
 #ifndef CJDB_CONTRACTS_HPP
 #define CJDB_CONTRACTS_HPP
 
-#include <cstdio>
 #include <string_view>
 #include <type_traits>
 
+#ifdef _MSC_VER
+	#define CJDB_PRETTY_FUNCTION __FUNCSIG__
+	#define CJDB_FORCE_INLINE __forceinline
+#else
+	#define CJDB_PRETTY_FUNCTION __PRETTY_FUNCTION__
+	#define CJDB_FORCE_INLINE [[gnu::always_inline]] inline
+#endif // _MSC_VER
+
+#ifndef CJDB_PRINT_ERROR
+	#ifdef CJDB_USE_STDIO
+		#include <cstdio>
+
+		namespace cjdb::contracts_detail {
+			struct print_error_fn {
+				CJDB_FORCE_INLINE void operator()(std::string_view const message) const noexcept
+				{
+					std::fwrite(message.data(), sizeof(char), message.size(), stderr);
```

Comment:
Why have you replaced `std::fprintf` with `std::fwrite`? My understanding is that `fwrite` is for binary files, and this library should be printing text.

Reply:
[`fwrite` is much faster than `fprintf`](https://stackoverflow.com/a/51152268/6797534). It is agnostic as to whether the file is opened in binary or text mode, but is [defined in terms of `fputc`](https://files.lhmouse.com/standards/ISO%20C%20N2176.pdf#subsection.7.21.8).

Reply:
Gotcha. `fwrite` appears to be an unformatted write, and `fprintf` is (as its name suggests) a formatted write.